### PR TITLE
virtualization: Add unit tests for virtualization.c

### LINKED ISSUE #25: Unit tests for virtualization.c

### PR DESCRIPTION
**Pull request by nhlfr**
Fixes #25

This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/24)

### CODE REVIEW

File: pkg/libvirttools/tests/test_virtualization.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+int
+main(int argc, char **argv)
+{
+	virConnectPtr conn;
+	gconstpointer gConn;
+	int result;
+
+	if (!(conn = virConnectOpen("test:///default"))) {
+		result = -1;
+		goto cleanup;
```
Comment (nebril) [inline pkg/libvirttools/tests/test_virtualization.c]: Just curious. Why `goto`? Can't you just put this in conditional block?

Comment (nhlfr) [inline pkg/libvirttools/tests/test_virtualization.c]: This is one of memory freeing methods - to create a cleanup block with return to not write all `free` statements separately on each error. It's widely used in libvirt itself as well.

However, I will try to switch to RAII later:
https://lwn.net/Articles/589433/
But I need to understand it better.

Comment (nebril) [inline pkg/libvirttools/tests/test_virtualization.c]: Cool thanks for explanation!

Comment (jellonek) [inline pkg/libvirttools/tests/test_virtualization.c]: IMO plain old `goto` usage is better than relaying on `__attribute__` extensions magic.

**Comment by pigmej** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/virtlet/24#-:-KSaAqvaBDSUhwxQ8YEt:bnfp4nl)*

**Pull request merged**